src/midi.js

```diff
@@ -5,15 +5,37 @@ const assert = require('assert');
 const jsmidgen = require('jsmidgen');
 
 /**
- * Take an array of note objects to generate a MIDI file in the same location as this method is called
+ * Take an array of note objects to asynchronously generate a MIDI file in the same location as this method is called
  * @param  {Array} notes    Notes are in the format: {note: ['c3'], level: 127, length: 64}
  * @param  {String} fileName If a filename is not provided, then `music.mid` is used by default
  * If `null` is passed for `fileName` bytes are returned instead of creating a file
+ *
+ * @param  {Object} callback Receives a callback method. Current signature for method is `err` in first param,
+ * in case something goes wrong.
+ * In case `fileName` is null, it will return a `err, bytes` signature for the function
+ *
  */
-const midi = (notes, fileName) => {
-	assert(Array.isArray(notes), 'You must provide an array of notes to write!');
+const midi = (notes, fileName, callback) => {
+	if (!callback || typeof callback !== 'function') {
```

Comment:
We should throw only if the `callback` was provided and it was not a function. Basically we shouldnt make it mandatory. In case a callback is not provided we should create one that simply consoles out that a MIDI file was created.

Reply:
Okay. But, if `fileName` is null, it will need to validate the presence of the callback as well. Without a `callback` and without a `fileName`, the user shouldn't be able to retrieve the `bytes` data

Reply:
If fileName is null, then it means they want bytes (not a file) in which case we wont need a callback.

Reply:
However, Did you mean the user wants to leave fileName to be `undefined` (basically they dont care bout the name of the file) but they do want a custom callback?